src/proof/proof-service.ts

```diff
@@ -213,11 +176,68 @@ export class ProofService implements IProofService {
     if (subjectGenesisDID.string() !== genesisDID.string()) {
       throw new Error('subject and auth profiles are not derived from the same did');
     }
-    const { inputs, vp } = await this.generateInputs(preparedCredential, genesisDID, proofReq, {
-      ...opts,
-      authProfileNonce,
-      credentialSubjectProfileNonce
-    });
+
+    const propertiesMetadata = await parseCredentialSubject(
+      proofReq.query.credentialSubject as JSONObject
+    );
+    if (!propertiesMetadata.length) {
+      throw new Error('no queries in zkp request');
+    }
+
+    const mtPosition = preparedCredential.credentialCoreClaim.getMerklizedPosition();
+
+    let mk: Merklizer | undefined;
+    if (mtPosition !== MerklizedRootPosition.None) {
+      mk = await preparedCredential.credential.merklize(this._ldOptions);
+    }
+
+    const context = proofReq.query['context'] as string;
```

Comment:
maybe we can parse proofReq.query to some internal struct ? 
context / type / ...